[Python] rectifyPoint is broken since OpenCV 2 Python API change

Just realized the Python `rectifyPoint()` method (in `PinholeCameraModel` inside `image_geometry` package) is broken, probably since the OpenCV 2 Python API change in this [commit](https://github.com/ros-perception/vision_opencv/commit/aca0824d3efb625d9a68786d093d864791bdac8d#diff-8fc927e9f451b1c3dae29275ac0cd56aR104).

It seems the `cv2.undistortPoints()` function in Python (OpenCV 2 API) behaves differently than in OpenCV 1 (they don't document it apparently). Possible solution: the input argument `src` has to be a [3d array](http://answers.opencv.org/question/28632/undistortpoints-with-python/) (?).

Just uncomment [this line](https://github.com/ros-perception/vision_opencv/blob/indigo/image_geometry/test/directed.py#L23) in the test and run it, and you will see it crash and give this error:

```
OpenCV Error: Assertion failed (src.isContinuous() && (src.depth() == CV_32F || src.depth() == CV_64F) && ((src.rows == 1 && src.channels() == 2) || src.cols*src.channels() == 2)) in undistortPoints, file /build/buildd/opencv-2.4.8+dfsg1/modules/imgproc/src/undistort.cpp, line 394
test_monocular ... ERROR!
ERROR: /build/buildd/opencv-2.4.8+dfsg1/modules/imgproc/src/undistort.cpp:394: error: (-215) src.isContinuous() &amp;&amp; (src.depth() == CV_32F || src.depth() == CV_64F) &amp;&amp; ((src.rows == 1 &amp;&amp; src.channels() == 2) || src.cols*src.channels() == 2) in function undistortPoints
```

I noticed I can get past this specific error if I comment out [line 102](https://github.com/ros-perception/vision_opencv/commit/aca0824d3efb625d9a68786d093d864791bdac8d#diff-8fc927e9f451b1c3dae29275ac0cd56aR102), but then I face another error:

```
cv2.error: /build/buildd/opencv-2.4.8+dfsg1/modules/imgproc/src/undistort.cpp:279: error: (-215) CV_IS_MAT(_src) && CV_IS_MAT(_dst) && (_src->rows == 1 || _src->cols == 1) && (_dst->rows == 1 || _dst->cols == 1) && _src->cols + _src->rows - 1 == _dst->rows + _dst->cols - 1 && (CV_MAT_TYPE(_src->type) == CV_32FC2 || CV_MAT_TYPE(_src->type) == CV_64FC2) && (CV_MAT_TYPE(_dst->type) == CV_32FC2 || CV_MAT_TYPE(_dst->type) == CV_64FC2) in function cvUndistortPoints
```

Notice this error is inside `cvUndistortPoints`, which is called by `undistortPoints` (the previous error), so it seems like the above-mentioned fix is at least fixing part of the problem (or is progress towards one).

--- a/image_geometry/src/image_geometry/cameramodels.py
+++ b/image_geometry/src/image_geometry/cameramodels.py
@@ -100,9 +100,8 @@ class PinholeCameraModel:
         """
 
         src = mkmat(1, 2, list(uv_raw))
-        src.resize((2, 1))
-        dst = src.copy()
-        cv2.undistortPoints(src, dst, self.K, self.D, self.R, self.P)
+        src.resize((1,1,2))
+        dst = cv2.undistortPoints(src, self.K, self.D, R=self.R, P=self.P)
         return dst[0,0]
 
     def project3dToPixel(self, point):

--- a/image_geometry/test/directed.py
+++ b/image_geometry/test/directed.py
@@ -20,7 +20,7 @@ class TestDirected(unittest.TestCase):
         print ci
         cam = PinholeCameraModel()
         cam.fromCameraInfo(ci)
-        #print cam.rectifyPoint((0, 0))
+        print cam.rectifyPoint((0, 0))
 
         print cam.project3dToPixel((0,0,0))
 

